File: test-code/src/main/java/com/example/testcode/member/Member.java
package com.example.testcode.member;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Entity
@Table(name = "member")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Member {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private long id;

  @Column(name = "email", nullable = false, updatable = false)
  private String email;

  @Column(name = "name", nullable = false)
  private String name;

  public Member(String email, String name) {
    this.email = email;
    this.name = name;
  }
}

File: kotlin-jpa/src/main/kotlin/com/example/kotlinjpa/member/Member.kt
package com.example.kotlinjpa.member

import javax.persistence.*


@Table(name = "member")
@Entity
class Member protected constructor() {

    @Id
    @GeneratedValue
    var id: Long = 0L

    @Column(name = "email", nullable = false, updatable = false)
    lateinit var email: String

    @Column(name = "name", nullable = false)
    lateinit var name: String


}
